src/components/bar-chart/bar-chart.jsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+import {
+    BarChart as BarChartWrapper, Bar, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer
+} from 'recharts';
+import ChartLegendContent from './chart-legend-content';
+import ChartTooltipContent from './chart-tooltip-content';
+
+const BarChart = ({
+    data,
+    dataKeys = [],
+    colors = [],
+    layout = "horizontal", // horizontal, vertical
+    stacked = false,
+    showXAxis = true,
+    showYAxis = true,
+    showTooltip = true,
+    tooltipIndicator = 'dot', // dot, line, dashed
+    tooltipLabelKey,
+    showLegend = false,
+    showCartesianGrid = true,
+    tickFormatter,
+    xAxisDataKey,
+    yAxisDataKey,
+    xAxisFontSize = 'sm', // sm, md, lg
+    xAxisFontColor = "#4B5563",
+    yAxisFontColor = "#4B5563",
+    chartWidth = 350,
+    chartHeight = 200,
+    borderRadius = 8,
+}) => {
+
+    const [width, setWidth] = useState(chartWidth);
+    const [height, setHeight] = useState(chartHeight);
+
+    useEffect(() => {
+        setWidth(chartWidth);
+        setHeight(chartHeight);
+    }, [chartWidth, chartHeight]);
```

Comment:
Can we directly use the width and height without using `useState` and  `useEffect`?

Reply:
At first, I did it directly using width and height, but then I had a problem in the storybook when changing width and height it does not render that changes. That's why I found solutions with useState and useEffect.

Reply:
Avoid this approach from the component side to avoid performance issues. If it’s not reflected in the Storybook, manage it from the Storybook or use the `key` prop to resolve this.